overlay.d/05core/usr/lib/dracut/modules.d/40ignition-ostree/ignition-ostree-transposefs.sh

```diff
@@ -139,13 +139,20 @@ should_autosave_rootfs() {
         return
     fi
     local agcount
+    # We want to run xfs_info on the unmounted filesystem, because actually just
+    # even mounting an XFS filesystem that has grown an excessive number of allocation groups
+    # can be very slow.
     eval $(xfs_info "${root_part}" | grep -o 'agcount=[0-9]*')
-    # Semi-arbitrarily chosen: this is roughly ~64G currently (based on initial
-    # ag sizing at build time) which seems like a good rootfs size at which to
+    # This is roughly ~700GiB currently (based on initial ag sizing at build time) which
+    # seems like a good rootfs size at which to
     # discriminate between "throwaway/short-lived systems" and "long-running
-    # workload systems". It's not like XFS performance is way worse at 128.
-    if [ "$agcount" -lt 128 ]; then
-        echo "Filesystem agcount is $agcount; skipping" >&2
+    # workload systems".  Specifically for e.g. OpenShift, it's below the default node
+    # sizes.
+    local threshold
+    threshold=400
+    echo "${root_part} agcount=$agcount" >&2
+    if [ "$agcount" -lt "${threshold}" ]; then
+        echo "agcount=$agcount is lower than threshold=${threshold}"
```

Comment:
The other echo in this function prints to stderr:

```suggestion
        echo "agcount=$agcount is lower than threshold=${threshold}" >&2
```

Also it might be nice to give one more piece of info:

```suggestion
        echo "agcount=$agcount is lower than the autosave-xfs threshold=${threshold}" >&2
```

Reply:
Personally, I always use `journalctl -o with-unit` which clearly shows the origin of things; it isn't reliable or scalable to try to prefix all the output inside a unit.  But sure, done anyways.

Reply:
> The other echo in this function prints to stderr:

I think these are normal conditions, so we should print to stdout.  Fixed the other cases.

Reply:
Yes, these are "normal" conditions, but we have to consider the way the function is used here. stdout gets swallowed by the caller and then the string is used in conditional logic.

Reply:
Oh wow you're right yes...writing correct shell script is just a mess.  Fixed